vistrails/gui/uvcdat/definedVariableWidget.py

```diff
@@ -449,14 +451,33 @@ def clear(self):
             d.show()
             self.ieds.append(d)
 
+    
     def trashVariable(self):
+        '''This method is used to delete all the selected variables. The variables are 
+           deleted only if they are not used to define other variables. They would however
+           be deleted if the defined variable is also selected for deletion''' 
+        selectedItems = self.varList.selectedItems()
+        varDeleted = 1
+        count = 1
+        while(varDeleted > 0):
+          varDeleted = 0 
+          for v in self.getSelectedDefinedVariables():
+              returnValue = self.deleteVariable(v.id,count)
+              if(returnValue):
+                varDeleted = varDeleted + 1
+        count = 2
         for v in self.getSelectedDefinedVariables():
-            self.deleteVariable(v.id)
+            self.deleteVariable(v.id,count)
```

Comment:
Second loop is to generate the popup errors of dependency if there are any
in the final list..

On Friday, March 6, 2015, Remi Rampin <notifications@github.com> wrote:

> In vistrails/gui/uvcdat/definedVariableWidget.py
> <https://github.com/UV-CDAT/VisTrails/pull/22#discussion_r25926962>:
>
> >          for v in self.getSelectedDefinedVariables():
> > -            self.deleteVariable(v.id)
> > +            self.deleteVariable(v.id,count)
>
> What is this second loop for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UV-CDAT/VisTrails/pull/22/files#r25926962>.
>